components/title-components/timer.tsx

```diff
@@ -1,10 +1,93 @@
+	let secondDelta = hackathonStart.getSeconds() - currentTime.getSeconds() + 59;
+	let minuteDelta = hackathonStart.getMinutes() - currentTime.getMinutes() + 59;
+	let hourDelta = hackathonStart.getHours() - currentTime.getHours() + 23;
+	let dayDelta = hackathonStart.getDate() - currentTime.getDate() - 1;
+	let monthDelta = hackathonStart.getMonth() - currentTime.getMonth();
+
+	if (hackathonStarted) {
+		secondDelta = hackathonEnd.getSeconds() - currentTime.getSeconds() + 59;
+		minuteDelta = hackathonEnd.getMinutes() - currentTime.getMinutes() + 59;
+		hourDelta = hackathonEnd.getHours() - currentTime.getHours() - 1;
+		dayDelta = hackathonEnd.getDate() - currentTime.getDate();
+		monthDelta = hackathonEnd.getMonth() - currentTime.getMonth();
+	}
+
+	if (hackathonEnded) {
+		secondDelta = 0;
+		minuteDelta = 0;
+		hourDelta = 0;
+		dayDelta = 0;
+		monthDelta = 0;
+	}
+
 	return (
-		<div>
-			{/* Text components */}
-			<div className="absolute top-[610px] left-0 flex items-center justify-center w-[calc(100vw-580px)]">
-				<h3 className="text-2xl xl:text-4xl">1 Month, 1 Day, 1 Hour, 1 Minute, 1 Second</h3>
+		<div className="w-full h-fit flex flex-col items-start justify-center">
+			{hackathonStarted && !hackathonEnded ? <h1 className="text-4xl font-bold mb-2">SUBMISSIONS DUE: </h1> : null}
+			{hackathonEnded ? <h1 className="text-4xl font-bold mb-2">THANKS FOR JOINING US! </h1> : null}
+			<div className="flex items-center justify-between w-full mb-4">
+				<Circle bgColor="bg-[#ef3a42]" textColor="text-white">
+					{monthDelta > 9 ? monthDelta : "0" + monthDelta}
+				</Circle>
+				<Circle bgColor="bg-[#f8a13a]" textColor="text-white">
+					{dayDelta > 9 ? dayDelta : "0" + dayDelta}
+				</Circle>
+				<Circle bgColor="bg-[#00a65c]" textColor="text-white">
+					{hourDelta > 9 ? hourDelta : "0" + hourDelta}
+				</Circle>
+				<Circle bgColor="bg-[#0058a9]" textColor="text-white">
+					{minuteDelta > 9 ? minuteDelta : "0" + minuteDelta}
+				</Circle>
+				<Circle bgColor="bg-[#b43c96]" textColor="text-white">
+					{secondDelta > 9 ? secondDelta : "0" + secondDelta}
+				</Circle>
 			</div>
+			<div className="flex items-center justify-between w-full">
+				<Circle bgColor="bg-[#ef3a42]" textColor="text-white">
+					M
+				</Circle>
+				<Circle bgColor="bg-[#f8a13a]" textColor="text-white">
+					D
+				</Circle>
+				<Circle bgColor="bg-[#00a65c]" textColor="text-white">
+					H
+				</Circle>
+				<Circle bgColor="bg-[#0058a9]" textColor="text-white">
+					M
+				</Circle>
+				<Circle bgColor="bg-[#b43c96]" textColor="text-white">
+					S
+				</Circle>
+			</div>
+		</div>
+	);
+}
+
+function Circle({ bgColor, textColor, children }: { bgColor: string; textColor: string; children: React.ReactNode }) {
+	return (
+		<div
+			className={`w-1/6 aspect-square rounded-full flex items-center justify-center ${bgColor} ${textColor} text-4xl sm:text-6xl desktop:text-5xl lg:text-6xl xl:text-7xl font-bold shadow-lg`}
```

Comment:
```suggestion
			className={w-10 h-10 sm:w-20 sm:h-20 md:w-24 md:h-24 lg:w-28 lg:h-28 aspect-square rounded-full flex items-center justify-center ${bgColor} ${textColor} text-4xl sm:text-6xl desktop:text-5xl lg:text-6xl xl:text-7xl font-bold shadow-lg`}
```
Setting a fixed width seems to fix the issue instead of a percentage width.

Reply:
![image](https://github.com/hack-rpi/HackRPI-Website-2024/assets/77216025/b666bd76-685b-41a7-a903-b4678595b612)
The dots become too small at certain widths